diff -up inkscape-1.0.1_2020-09-07_3bc2e813f5/src/ui/tool/node.cpp.omv~ inkscape-1.0.1_2020-09-07_3bc2e813f5/src/ui/tool/node.cpp
--- inkscape-1.0.1_2020-09-07_3bc2e813f5/src/ui/tool/node.cpp.omv~	2020-09-08 16:35:36.754216519 +0200
+++ inkscape-1.0.1_2020-09-07_3bc2e813f5/src/ui/tool/node.cpp	2020-09-08 16:35:45.472322927 +0200
@@ -8,6 +8,7 @@
  */
 
 #include <iostream>
+#include <atomic>
 #include <stdexcept>
 #include <boost/utility.hpp>
 
