--- a/src/ui/dialog/filedialogimpl-gtkmm.cpp	Thu Jan 15 10:14:12 1970
+++ b/src/ui/dialog/filedialogimpl-gtkmm.cpp	Thu Jan 15 10:14:12 1970
@@ -141,13 +141,12 @@
 
     //This should remove it from the box, and free resources
     if (viewerGtk)
-        gtk_widget_destroy(viewerGtk);
-
-    viewerGtk  = sp_svg_view_widget_new(doc);
-    GtkWidget *vbox = (GtkWidget *)gobj();
-    gtk_box_pack_start(GTK_BOX(vbox), viewerGtk, TRUE, TRUE, 0);
-    gtk_widget_show(viewerGtk);
+        Gtk::Container::remove(*viewerGtk);
 
+    viewerGtk  = Glib::wrap(sp_svg_view_widget_new(doc));
+    Gtk::VBox *vbox = Glib::wrap(gobj());
+    vbox->pack_start(*viewerGtk,  TRUE, TRUE, 0);
+    viewerGtk->show();
     return true;
 }
 
--- a/src/ui/dialog/filedialogimpl-gtkmm.h	Thu Jan 15 10:14:12 1970
+++ b/src/ui/dialog/filedialogimpl-gtkmm.h	Thu Jan 15 10:14:12 1970
@@ -121,7 +121,7 @@
     /**
      * The sp_svg_view widget
      */
-    GtkWidget *viewerGtk;
+    Gtk::Widget *viewerGtk;
 
     /**
      * are we currently showing the "no preview" image?
