diff -up inkscape-0.92.4/src/extension/internal/pdfinput/pdf-parser.cpp.omv~ inkscape-0.92.4/src/extension/internal/pdfinput/pdf-parser.cpp
--- inkscape-0.92.4/src/extension/internal/pdfinput/pdf-parser.cpp.omv~	2019-05-08 16:41:08.722441361 +0200
+++ inkscape-0.92.4/src/extension/internal/pdfinput/pdf-parser.cpp	2019-05-08 16:42:06.873032684 +0200
@@ -426,7 +426,7 @@ void PdfParser::parse(Object *obj, GBool
 	error(errInternal, -1, "Weird page contents");
     	return;
   }
-  parser = new Parser(xref, new Lexer(xref, obj), gFalse);
+  parser = new Parser(xref, obj, gFalse);
   go(topLevel);
   delete parser;
   parser = NULL;
