diff -up inkscape-1.4_2024-10-09_e7c3feb100/src/extension/internal/pdfinput/pdf-parser.cpp.omv~ inkscape-1.4_2024-10-09_e7c3feb100/src/extension/internal/pdfinput/pdf-parser.cpp
--- inkscape-1.4_2024-10-09_e7c3feb100/src/extension/internal/pdfinput/pdf-parser.cpp.omv~	2025-01-06 02:32:34.804381764 +0100
+++ inkscape-1.4_2024-10-09_e7c3feb100/src/extension/internal/pdfinput/pdf-parser.cpp	2025-01-06 02:32:41.635256553 +0100
@@ -2397,7 +2397,7 @@ void PdfParser::doImage(Object * /*ref*/
     // get info from the stream
     bits = 0;
     csMode = streamCSNone;
-    str->getImageParams(&bits, &csMode);
+    str->getImageParams(&bits, &csMode, nullptr);
     
     // get stream dict
     dict = str->getDict();
