diff -up inkscape-1.2.1_2022-07-14_9c6d41e410/src/extension/internal/pdfinput/pdf-parser.cpp.omv~ inkscape-1.2.1_2022-07-14_9c6d41e410/src/extension/internal/pdfinput/pdf-parser.cpp
--- inkscape-1.2.1_2022-07-14_9c6d41e410/src/extension/internal/pdfinput/pdf-parser.cpp.omv~	2022-09-01 23:13:40.465269327 +0200
+++ inkscape-1.2.1_2022-07-14_9c6d41e410/src/extension/internal/pdfinput/pdf-parser.cpp	2022-09-01 23:17:24.674935584 +0200
@@ -685,19 +685,19 @@ void PdfParser::opConcat(Object args[],
 // TODO not good that numArgs is ignored but args[] is used:
 void PdfParser::opSetDash(Object args[], int /*numArgs*/)
 {
-  double *dash = nullptr;
+  std::vector<double> dash;
 
   Array *a = args[0].getArray();
   int length = a->getLength();
   if (length != 0) {
-    dash = (double *)gmallocn(length, sizeof(double));
+    dash.resize(length);
     for (int i = 0; i < length; ++i) {
       Object obj;
       dash[i] = _POPPLER_CALL_ARGS_DEREF(obj, a->get, i).getNum();
       _POPPLER_FREE(obj);
     }
   }
-  state->setLineDash(dash, length, args[1].getNum());
+  state->setLineDash(std::move(dash), args[1].getNum());
   builder->updateStyle(state);
 }
 
diff -up inkscape-1.2.1_2022-07-14_9c6d41e410/src/extension/internal/pdfinput/svg-builder.cpp.omv~ inkscape-1.2.1_2022-07-14_9c6d41e410/src/extension/internal/pdfinput/svg-builder.cpp
--- inkscape-1.2.1_2022-07-14_9c6d41e410/src/extension/internal/pdfinput/svg-builder.cpp.omv~	2022-09-01 23:11:09.906870365 +0200
+++ inkscape-1.2.1_2022-07-14_9c6d41e410/src/extension/internal/pdfinput/svg-builder.cpp	2022-09-01 23:17:58.499587513 +0200
@@ -388,15 +388,13 @@ void SvgBuilder::_setStrokeStyle(SPCSSAt
     sp_repr_css_set_property(css, "stroke-miterlimit", os_ml.str().c_str());
 
     // Line dash
-    double *dash_pattern;
-    int dash_length;
     double dash_start;
-    state->getLineDash(&dash_pattern, &dash_length, &dash_start);
-    if ( dash_length > 0 ) {
+    const std::vector<double> dash_pattern = state->getLineDash(&dash_start);
+    if ( dash_pattern.size() > 0 ) {
         Inkscape::CSSOStringStream os_array;
-        for ( int i = 0 ; i < dash_length ; i++ ) {
+        for ( int i = 0 ; i < dash_pattern.size() ; i++ ) {
             os_array << dash_pattern[i];
-            if (i < (dash_length - 1)) {
+            if (i < (dash_pattern.size() - 1)) {
                 os_array << ",";
             }
         }
